### ISSUE TITLE
Problems reading boolean values for traits using comments

### ISSUE THREAD

**Issue by Tarjei400** [apache/camel-k#4493]
Version: 1.12.1

I am trying to run integration example
```                               
// camel-k: language=kotlin
// camel-k: trait=quarkus.package-type=fast-jar
// camel-k: trait=quarkus.package-type=native
// camel-k: trait=prometheus.enabled=true

// Write your routes here, for example:
from("timer:kotlin?period=1000")
  .routeId("kotlin")
  .setBody()
    .simple("Hello Camel K from \${routeId}!")
  .to("log:info")
  ```
  
  Not sure if this is recommended way of running integrations, but noticed it has various problems when reading trait configs. Without trait configs, deployment runs properly.
  
  @edit seems that issue happens when I try to run it in k3s, on GCP everything is fine.

**Comment by gansheer** [comment]
Hello,

Can you give us more details on your issues, like which traits are not config don't work or some error logs if you have some ?

**Comment by squakez** [comment]
> edit seems that issue happens when I try to run it in k3s, on GCP everything is fine.

Still, it would be good to know what are the errors so we can advice how to fix them.

**Comment by Tarjei400** [comment]
Actually, I ended up having a bit of a meltdown with my K3s setup, so I shifted everything over to Google Cloud Platform (GCP) instead. Happily, everything seems to be smooth sailing over there, no flag troubles at all.

Back when I was still on K3s, I remember running into problems when I tried to read boolean flags. It didn't matter if I had them tucked into comments or if I tried passing them straight to kamel run as a trait parameter. Sorry that I can't offer more specifics than that at the moment.

Given that I've moved away from K3s and I'm not seeing the same problem on GCP, it's probably best to close this issue. If I run into any more problems or figure out anything else that might be helpful, I'll be sure to open a new issue.
